### ISSUE TITLE
Configuration not being loaded

### ISSUE THREAD

**Issue by chippyash** [kenjis/ci-phpunit-test#151]
Plain vanilla install as per readme.

```
cd app/tests
../../vendor/bin/phpunit
```
I get 

> PHP Notice:  Undefined index: log_path in /.../system/core/Log.php on line 118

If I put a `var_dump($config);exit;` in at L117 of system/core/Log.php, sure enough, the entry is missing, even though it is declared in app/config/config.php.  In fact the only entry in $config is 'base_url'.

Any ideas?
Thanks

**Comment by kenjis** [comment]
No idea.
Do you see the CI welcome page via your browaser (without the notice)?

I can't reproduce the notice.

Installation:
```
$ curl -L https://github.com/bcit-ci/CodeIgniter/archive/3.1.2.zip -o CodeIgniter-3.1.2.zip
$ unzip CodeIgniter-3.1.2.zip
$ curl -L https://github.com/kenjis/ci-phpunit-test/archive/v0.13.0.zip -o ci-phpunit-test-0.13.0.zip
$ unzip ci-phpunit-test-0.13.0.zip
$ cp -r ci-phpunit-test-0.13.0/application/tests/ CodeIgniter-3.1.2/application/tests
$ cd CodeIgniter-3.1.2
$ curl -L https://phar.phpunit.de/phpunit.phar -o phpunit.phar
```

Running tests:
```
$ php phpunit.phar -c application/tests/
PHPUnit 5.6.5 by Sebastian Bergmann and contributors.

Error:         No code coverage driver is available

...                                                                 3 / 3 (100%)

Time: 330 ms, Memory: 10.00MB

OK (3 tests, 3 assertions)
```

**Comment by REJack** [comment]
I had a similar error, if i remind me right i think your config.php doesn't contain all config vars that CI3 needs.

**Comment by kenjis** [comment]
If you still have this problem, feel free to reopen.

**Comment by aderowbotham** [comment]
I have this problem with CI 3.1.9

    php phpunit.phar -c application/tests/

    Notice: Undefined index: log_path in /path/to/deploy/system/core/Log.php on line 127

**Comment by aderowbotham** [comment]
I have this problem with CI 3.1.9

Running `php phpunit.phar -c application/tests/` I get notices and errors. It seems **constants.php** is never loaded

Here is the output with the first stack trace:

php phpunit.phar -c application/tests/

    Notice: Undefined index: log_path in /path/to/deploy/system/core/Log.php on line 127

    Call Stack:
        0.0435     857536   1. {main}() /path/to/deploy/phpunit.phar:0
        0.1654    9031552   2. PHPUnit\TextUI\Command::main() /path/to/deploy/phpunit.phar:570
        0.1668    9031664   3. PHPUnit\TextUI\Command->run() phar:///path/to/deploy/phpunit.phar/phpunit/TextUI/Command.php:148
        0.1671    9031664   4. PHPUnit\TextUI\Command->handleArguments() phar:///path/to/deploy/phpunit.phar/phpunit/TextUI/Command.php:159
        0.2041    9035664   5. PHPUnit\TextUI\Command->handleBootstrap() phar:///path/to/deploy/phpunit.phar/phpunit/TextUI/Command.php:788
        0.2041    9035664   6. PHPUnit\Util\Fileloader::checkAndLoad() phar:///path/to/deploy/phpunit.phar/phpunit/TextUI/Command.php:991
        0.2051    9035968   7. PHPUnit\Util\Fileloader::load() phar:///path/to/deploy/phpunit.phar/phpunit/Util/Fileloader.php:48
        0.2061    9061144   8. include_once('/path/to/deploy/application/tests/Bootstrap.php') phar:///path/to/deploy/phpunit.phar/phpunit/Util/Fileloader.php:64
        0.2099    9092600   9. CIPHPUnitTest::init() /path/to/deploy/application/tests/Bootstrap.php:386
        0.2978    9855624  10. CIPHPUnitTest::replaceHelpers() /path/to/deploy/vendor/kenjis/ci-phpunit-test/application/tests/_ci_phpunit_test/CIPHPUnitTest.php:89
        0.2978    9855624  11. CIPHPUnitTest::loadHelper() /path/to/deploy/vendor/kenjis/ci-phpunit-test/application/tests/_ci_phpunit_test/CIPHPUnitTest.php:222
        0.2978    9855624  12. config_item() /path/to/deploy/vendor/kenjis/ci-phpunit-test/application/tests/_ci_phpunit_test/CIPHPUnitTest.php:228
        0.2978    9855648  13. get_config() /path/to/deploy/vendor/kenjis/ci-phpunit-test/application/tests/_ci_phpunit_test/replacing/core/Common.php:408
        0.2984    9907456  14. require('/path/to/deploy/application/config/config.php') /path/to/deploy/vendor/kenjis/ci-phpunit-test/application/tests/_ci_phpunit_test/replacing/core/Common.php:350
        0.2984    9915544  15. _error_handler() /path/to/deploy/application/config/config.php:110
        0.3010    9939040  16. CI_Exceptions->log_exception() /path/to/deploy/system/core/Common.php:618
        0.3010    9939296  17. log_message() /path/to/deploy/system/core/Exceptions.php:105
        0.3010    9939320  18. load_class() /path/to/deploy/vendor/kenjis/ci-phpunit-test/application/tests/_ci_phpunit_test/replacing/core/Common.php:204
        0.3015    9967376  19. CI_Log->__construct() /path/to/deploy/vendor/kenjis/ci-phpunit-test/application/tests/_ci_phpunit_test/replacing/core/Common.php:111

then I get 4 further errors (stack trace removed here)

    Notice: Undefined index: log_path in /path/to/deploy/system/core/Log.php on line 127
    Warning: mkdir(): Invalid path in /path/to/deploy/system/core/Log.php on line 131
    Notice: Undefined index: log_threshold in /path/to/deploy/system/core/Log.php on line 138
    Notice: Undefined index: log_threshold in /path/to/deploy/system/core/Log.php on line 142

then I get some application-specific errors relating to constants not being defined. These are all defined in my config/constants.php. constants.php is not being loaded, possibly due to the problems in Log.php

    A PHP Error was encountered

    Severity:    Warning
    Message:     Use of undefined constant ENV_PRODUCTION - assumed 'ENV_PRODUCTION' (this will throw an Error in a future version of PHP)
    Filename:    /path/to/deploy/application/config/config.php
    Line Number: 110

**Comment by aderowbotham** [comment]
This system seems to be loading **config.php** before **constants.php**. This is the reverse to how CI loads these files. Our application uses lots of constants so this file must be loaded first.

    A PHP Error was encountered

    Severity:    Warning
    Message:     Use of undefined constant CI_ENCRYPTION_KEY - assumed 'CI_ENCRYPTION_KEY' (this will throw an Error in a future version of PHP)
    Filename:    /path/to/deploy/application/config/config.php
    Line Number: 497

    A PHP Error was encountered

    Severity:    Warning
    Message:     Use of undefined constant ENV_DEVELOPMENT - assumed 'ENV_DEVELOPMENT' (this will throw an Error in a future version of PHP)
    Filename:    /path/to/deploy/application/config/config.php

**Comment by kenjis** [comment]
@aderowbotham I've confirmed that `config.php` is loaded before `constants.php`.
You are right. This behavior is the reverse to how CI does.

**Comment by kenjis** [comment]
@aderowbotham I fixed `master` and `2.x` branches.

**Comment by aderowbotham** [comment]
@kenjis We are installing from Composer currently using **^0.18.0**. There is no later release than 0.18.0 so this does not get the fix.

Is it possible to make a new release to update from 0.18.0?

**Comment by kenjis** [comment]
@aderowbotham The fix will be included in v0.19.0 which is not released.

You can use developing version (`master` branch) like this:
```
"kenjis/ci-phpunit-test": "1.0.x@dev",
```

**Comment by MGParisi** [comment]
It would be nice to have a fix.... As it appears we hit the same issue.

**Comment by kenjis** [comment]
Fixed in v0.19.0.
